File: 백준/Gold/1700. 멀티탭 스케줄링/멀티탭 스케줄링.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        int[] order = new int[k];
        st = new StringTokenizer(br.readLine());
        for(int i=0;i<k;i++){
            order[i] = Integer.parseInt(st.nextToken());
        }
        List<Integer> plug  = new ArrayList<>();
        int res = 0;

        for(int i=0;i<k;i++){
            int current = order[i];
            if(plug.contains(current)){
                continue;
            }
            if(plug.size()<n){
                plug.add(current);
                continue;
            }
            int farthest = -1;
            int delete = -1;
            for(int j=0;j<plug.size();j++){
                int device = plug.get(j);
                int next;
                next = finduse(order,i+1,device);
                if(next==-1){
                    delete = j;
                    break;
                }
                if(next>farthest){
                    farthest = next;
                    delete = j;
                }
            }
            plug.set(delete, current);
            res++;

        }
        System.out.println(res);
    }
    private static int finduse(int[] order, int start, int device){
        for(int i=start;i<order.length;i++){
            if(order[i]==device){
                return i;
            }
        }
        return -1;
    }

    }

File: 백준/Gold/1700. 멀티탭 스케줄링/멀티탭 스케줄링.py
import sys
input = sys.stdin.readline


n,k = map(int,input().split())
order = list(map(int,input().split()))
plug = []
res = 0
for i in range(k):
    if order[i] in plug:
        continue
    if len(plug)<n:
        plug.append(order[i])
        continue
    farthset =-1
    delete = -1
    for j in range(len(plug)):
        try:
            next_use = order[i+1:].index(plug[j])#다음으로 사용할 위치
        except ValueError:
            next_use = float('inf') #앞으로 사용 ㄴㄴ
        if next_use>farthset:
            farthset = next_use
            delete = j
    plug[delete] = order[i]
    res+=1
print(res)